src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -1,16 +1,44 @@
+import { BigNumber } from '@ethersproject/bignumber'
 import { useCallback } from 'react'
 import { useClearQuote, useUpdateQuote } from 'state/price/hooks'
 import { getCanonicalMarket, registerOnWindow } from 'utils/misc'
 import { FeeQuoteParams, getFeeQuote, getPriceQuote } from 'utils/operator'
+import { FeeInformation, PriceInformation } from '../state/price/reducer'
 
 export interface RefetchQuoteCallbackParmams {
   quoteParams: FeeQuoteParams
   fetchFee: boolean
+  previousFee?: FeeInformation
 }
 
-function handleError(err: any): undefined {
-  console.error('Error fetching price/fee', err)
-  return undefined
+async function getQuote({
+  quoteParams,
+  fetchFee,
+  previousFee
+}: RefetchQuoteCallbackParmams): Promise<[PriceInformation, FeeInformation]> {
+  const { sellToken, buyToken, amount, kind, chainId } = quoteParams
+  const { baseToken, quoteToken } = getCanonicalMarket({ sellToken, buyToken, kind })
+
+  // Get a new fee quote (if required)
+  const feePromise =
+    fetchFee || !previousFee ? getFeeQuote({ chainId, sellToken, buyToken, amount, kind }) : previousFee
+
+  // Get a new price quote
+  let exchangeAmount
+  if (kind === 'sell') {
+    // Sell orders need to deduct the fee from the swapped amount
+    exchangeAmount = BigNumber.from(amount)
+      .sub((await feePromise).amount)
```

Comment:
this won't work with changes made in #619 as the endpoints can now return an error response useful to detecting unsupported tokens. 

maybe we merge this in now and then adjust it if/when we merge #619 using the typecheck functions

Reply:
Ok! let me me know how is best to consolidate

Reply:
i will do it it's easier, let's just merge this asap